Sources/LightweightCharts/LightweightChartsModels/MouseEventParams.swift

```diff
@@ -90,11 +169,14 @@ extension EventPrices: Codable {
     
     public init(from decoder: Decoder) throws {
         let container = try decoder.singleValueContainer()
-        if let barPrice = try? container.decode(BarPrice.self) {
-            self = .barPrice(barPrice)
-        } else if let barPrices = try? container.decode(BarPrices.self) {
-            self = .barPrices(barPrices)
-        } else {
+        let barData = try? container.decode(BarData.self)
+        let lineData = try? container.decode(LineData.self)
+        
+        if (lineData?.value != nil){
+            self = .lineData(lineData!)
```

Comment:
it'll be probably a bit better like this:

```swift
if let lineData = try? container.decode(LineData.self), lineData.value != nil {
    self = .lineData(lineData)
} else {
```

Reply:
done